### ISSUE TITLE
Create a list of user stories

### ISSUE THREAD

**Issue by cwmeijer** [dianna-ai/dianna#102]
As a <developer/domain scientist/ML engineer/...> I want <do something> in order to <gain insight for instance>.

**Comment by elboyran** [comment]
[Blog post](https://www.atlassian.com/agile/project-management/user-stories) about user stories in Agile development.

**Comment by elboyran** [comment]
Unfortunately, no responses from Sonja (as a link to an interested PI) nor Elisabeth (SSI-ML). Yang will remind Sonja, I will write one on Elisabeth's behalf and/or maybe Leon can also ask if soon to meet in the SSI-ML context?

**Comment by elboyran** [comment]
An example user story based on my understanding of the wishes of Elisabeth Lodder from AMC:

**As a domain scientist (Elisabeth) _I want to have_ an XAI tool which works on time series _in order to_ verify (from a medical expert view point) an DNN-based detection of a specific medical syndrome from EEG signals.**

**Comment by elboyran** [comment]
Generated from suggestions by Nico:

**As a domain scientist (a chemist) _I want to have_ an XAI tool which works on graph structures (like molecules) _in order to_ understand the importance of a functional group in the molecular structure.**

**Comment by cwmeijer** [comment]
These are helpful stories! They tend to be a bit general. I'll make some suggestions that would make them more specific:
Elisabeth:
As a domain scientist I want to see heat maps of my time series data (EEG signals) wrt classes in order to an DNN-based detection of a specific medical syndrome.
As a domain scientist I want to edit (draw?) my time series data (EEG signals) in order to explore the change in behavior of an DNN-based detection of a specific medical syndrome.
As a domain scientist I want to zoom in and out in my time series heat maps in order to inspect details.
As a domain scientist I want to export my heat maps to a convenient format like csv/json/... in order to analyze the data further with my own tools.

Nico:
As a domain scientist I want to view my network and have edges thickness / color coded wrt importance on classification in order to understand the reasons of the prediction by a graph DNN.
As a domain scientist I want to filter my network visualization using <this or that> filters in order to inspect the map with less clutter.

Do you think this is what they want? What else do they want? Obviously they want the tool to do all of the above and more, but what is really important to them?

**Comment by elboyran** [comment]
I don't have experience with user stories, so yours look fine. What I was aiming for was the high level scientific questions they seem to want to answer. I guess it's then up to us to convert those into technical requirements and your stories seem to be doing that ;-). Nico gave the functional group as a concrete example, so maybe we can ask him if these definitions of user stories make sense?

**Comment by cwmeijer** [comment]
Yes, ask them to look at these and think about some more that are specific. After that, it is important to know from them which ones are really important and which ones are less important but convenient etc. Otherwise I'm sure they will just agree with whatever complete list we will show to them. This would make it very hard to make any minimal viable product.

I would expect that a feature like the ability to draw on eeg data to alter it (point 2), would be something fairly low on the priority list. It is there also for inspiration to create more user stories like it. But that is up to you, as a product owner, to figure out, with help from our stakeholders, and your team of course :-)

**Comment by elboyran** [comment]
Feedback by Simona (Post-doc for Elisabeth):

> An alternative can be the following:
> 
> As a domain scientist, I want to have an XAI tool that works on time series in order to identify the importance of each data point of the time series given by the DNN-based prediction model for the detection of a specific medical syndrome.  
> 
> I have done this for the LQTS project, using a modified version of the CAM-Grad technique. 
> If there are alternative methods that can be used to understand how important each data point is, we can compare the results of the different methods. 
> The importance can be a score, and it can be afterward color-coded and visualized on the time series.

**Comment by elboyran** [comment]
Generated by me from a finalist proposal for Loretnz eScience workshop:

**As a climate scientist (Dim) _I want to have_ an XAI tool which works on time series climate data for seasonal forecasts _in order to_ gain trust and understand the sources of predictability**

**Comment by elboyran** [comment]
Generated by me from a finalist proposal for Loretnz eScience workshop:

**As a climate scientist (Dim) _I want to have_ an XAI tool which works on time series climate data for seasonal forecasts _in order to_ gain trust and understand the sources of predictability**

**Comment by elboyran** [comment]
> 
> 
> Generated from suggestions by Nico:
> 
> **As a domain scientist (a chemist) _I want to have_ an XAI tool which works on graph structures (like molecules) _in order to_ understand the importance of a functional group in the molecular structure as predicted by a graph DNN.**

Update:

**As a computational biologist working on protein-protein interactions, I want a XAI tool on graph structures that helps me understanding the importance of atoms/residues  as well as interactions between pairs of atoms/residues in the forming of a stable protein-protein interface.**
